_posts/2020-04-14-s3-access.md

```diff
@@ -0,0 +1,212 @@
+
+  It's usually reasonably obvious what is going on, except for the two things - "ListAllBuckets" and "ListBucket". The former lists all the buckets in the account, the latter lists all of the objects in a particular bucket. Yes, it's easy to mix up.
+
+- Effect
+
+  The effect is the result of various conditions matching the rest of the policy. We can do two things - we can either ALLOW things to happen or DENY them.
+
+- Principal
+
+  The principal is the identity of the 'thing' performing the action. For example, this lets us to block Mark from reading an object but allows Alice to read it. Users, roles, services, accounts, all of these things are 'principals'. Here are some examples.
+
+  Give access to the AWS Config service.
+
+  ```json
+  {
+
+      "Principal": {
+          "Service": "config.amazonaws.com"
+      }
+  }
+  ```
+
+  Give access to all accounts within AWS.
+  ```json
+  {
+      "Principal": {
+          "AWS": "*"
+      }
+  }
+  ```
+
+  Give access to two accounts - the following two examples are equivalent.
+  Important! Note the usage of 'root' doesn't mean root user, or root account. It just means the account is free to delegate permission as it sees fit.
+  ```json
+  {
+      "Principal": {
+         "AWS": [ "0123456789", "123456790" ]
+      }
+  }
+  ```
+
+  ```json
+  {
+      "Principal": {
+         "AWS": [ "arn:aws:iam::0123456789:root", "arn:aws:iam::123456790:root" ]
+      }
+  }
+  ```
+
+  Give access to everyone in the world
+  ```json
+  {
+      "Principal": "*"
+  }
+  ```
+
+  Give access to a specific role within an account
+  ```json
+  {
+      "Principal": {
+          "AWS": "arn:aws:iam::1234567890:role/my-cool-role"
+      }
+  }
+  ```
+
+  Note that the only time wildcards work in a policy is when they are specified as the only thing in a value - "*" will work, but "arn:aws:iam::1234567890:role/*" will not.
+
+- Condition
+
+  Condition statements allow us to further restrict access based on various circumstantial attributes. Maybe we block Mark all the time, except for when Mark is accessing the bucket from a particular office.
+
+- Canonical ID
+
+  Canonical ID's pop up in a few places but you are more likely to see them in ACLs than anywhere else. They usually identify what accounts may have access to certain buckets.
+
+- Resource Policy & IAM Policies
+
+  Resource policies are access policies that apply to 'resources' as opposed to 'identities' (like IAM Policies). Resource policies are most notable in that they include a 'principal' statement to narrow down the identity accessing the resource. Identity policies do not have a principal statement because they attached to identities - the principal is therefore implied.
+
+  Note: As dicussed before, principal statements can not include wildcard ('*') statements.
```

Comment:
instead of brackets, use backticks ` for syntax highlighting so that the user is clear on which part needs to be in a policy

Reply:
I've tried to stop doing that inline because I feel like it makes the text harder to read.